staging/src/k8s.io/api/coordination/v1/types.go

```diff
@@ -58,6 +66,10 @@ type LeaseSpec struct {
 	// holders.
 	// +optional
 	LeaseTransitions *int32 `json:"leaseTransitions,omitempty" protobuf:"varint,5,opt,name=leaseTransitions"`
+	// Strategy indicates the strategy for picking the leader for coordinated leader election
+	// +optional
+	// +default="NoCoordination"
+	Strategy *CoordinatedStrategy `json:"strategy,omitempty" protobuf:"bytes,6,opt,name=strategy"`
```

Comment:
As string, don't think this needs to be a pointer.

Reply:
Do we consider this an alpha field?

Reply:
This is adding a field to a v1 type and I see the general practice is to make them pointers to ensure backwards compatibility. Prior art: https://github.com/kubernetes/kubernetes/commit/36f30bc68953a61beb18e5bdb41942e6b0993b9f#diff-5ffbebb40a29f8e491f4ff1e07c618eb3edcc0d5f00a1a71445272d6767d7883.

Reply:
Interesting. Pointers, not only optional I assume. Do you have a reference where to read about it?

Reply:
All newly-added fields to already-GA APIs *must* be optional.  Proto serializes non-pointer fields.  We don't want to serialize pre-GA fields which are disabled.  Ergo newly-added fields to already-GA APIs *must* be pointers.

Consider:
1) a `string` field with `omitempty` and a value of `""`
  - JSON will elide this
  - protobuf will render this
2) a `*string` field with `omitempty` and a value of '""`
  - JSON will render this
  - protobuf will render this
3) a `*string` field with `omitempty` and a nil value
  - JSON will elide this
  - protobuf will elide this

For newly added fields in GA APIs, this could matter.  Even if we do the usual `dropDisabledFields()` thing, an empty string (tag ID + 0 bytes of data) will appear in the serialized proto, whereas a nil `*string` will not.

Blame @liggitt for pointing this out.